concrete/src/Service/Detector/HTTP/ApacheDetector.php

```diff
@@ -29,19 +29,37 @@ public function __construct(Request $request)
     public function detect()
     {
         $result = null;
-        if ($result === null && $this->request->server->has('SERVER_SOFTWARE')) {
-            $result = $this->detectFromServer($this->request->server->get('SERVER_SOFTWARE'));
+        if (($result === null || $result === '') && $this->request->server->has('SERVER_SOFTWARE')) {
+            $version = $this->detectFromServer($this->request->server->get('SERVER_SOFTWARE'));
+            if ($version !== null) {
+                $result = $version;
+            }
         }
-        if ($result === null && function_exists('apache_get_version')) {
-            $result = $this->detectFromSPL(@apache_get_version());
+        if (($result === null || $result === '') && function_exists('apache_get_version')) {
+            $version = $this->detectFromSPL(@apache_get_version());
```

Comment:
> @mlocati I mean because it basically duplicates this block

Because I am stupid? :wink: -> https://github.com/concrete5/concrete5/pull/5481/commits/81c5aa592a96dcce72c8cf7fe3f976300fe5d3d4